Automattic/wp-calypso
# Post Editor: fix default values of discussion options when editing new post

This PR adds a few more site options to request when fetching the sites list. This fixes a bug when editing a new post in the editor.

**First commit: Extract lists of site fields and options to a module with constants**
This patch deduplicates the list of site fields and options to fetch into `state/sites/constants` module. Until now, they were duplicated in state/jetpack-connect and state/sites and slowly getting out of sync. For example, `site_goals` was added only to one place in #22171.

Also, the list is now an array with one item per line. Makes for a much more readable diffs when a field is added or removed.

**Second commit: start fetching options for default values of post comments/pingback settings**
Request the `default_ping_status` and `default_comment_status` options when requesting sites: they are used as defaults when editing a new post.

**How to test:**
First, go to the Discussion settings of your site and edit the default values for "Allow link notifications from other blogs" and "Allow people to post comments": 
<img width="548" alt="post-discussion-settings" src="https://user-images.githubusercontent.com/664258/37971387-1a0a47c0-31d6-11e8-8fb5-3adc113cdef3.png">

Then, start editing a new blog post on that site and verify in the sidebar UI that the defaults are really applied:
<img width="272" alt="post-discussion-good" src="https://user-images.githubusercontent.com/664258/37971495-5fb39d12-31d6-11e8-8ca5-606400f11bb5.png">

Before this PR, the defaults were always empty:
<img width="264" alt="post-discussion-bad" src="https://user-images.githubusercontent.com/664258/37971540-798d760e-31d6-11e8-8654-6fda72cad99c.png">
And were updated to the correct values only after first autosave. Which is triggered only after you enter a post title and start typing for a while.
